src/hooks/resources/device.ts

```diff
@@ -494,3 +509,64 @@ addDeleteHookForDependents('device', [
 	['service_install', 'device'],
 	['gateway_download', 'is_downloaded_by__device'],
 ]);
+
+function checkSupervisorReleaseUpgrades(
+	api: PinejsClient,
+	deviceIds: number[],
+	newSupervisorReleaseId: number,
+) {
+	return api
+		.get({
+			resource: 'supervisor_release',
+			id: newSupervisorReleaseId,
+			options: {
+				$select: 'supervisor_version',
+			},
+		})
+		.then((newSupervisorRelease: AnyObject) => {
+			if (newSupervisorRelease == null) {
+				throw new BadRequestError(
+					`Could not find a supervisor release with this ID ${newSupervisorReleaseId}`,
+				);
+			}
+			const newSupervisorVersion = newSupervisorRelease.supervisor_version;
+			return Bluebird.map(deviceIds, deviceId => {
+				return api
```

Comment:
That can be a single request fetching all affected devices and you can then map over the results.